things/api.py

```diff
@@ -314,6 +314,9 @@ def tags(title=None, include_items=False, **kwargs):
 # Utility API functions derived from above
 # --------------------------------------------------
 
+def search(querystr: str, **kwargs):
```

Comment:
Maybe call the parameter `query` for simplicity?

Reply:
For me, it would help to have documentation and examples on this parameter. What types of query strings are expected and supported here?

Reply:
> Maybe call the parameter `query` for simplicity?

Main reason was consistency, as we can't use `query` in `database.py`. However, changed it in `api.py`.

> For me, it would help to have documentation and examples on this parameter. What types of query strings are expected and supported here?
 
Added more comments. I noticed that, imho, we should use [Python Type Hinting](https://docs.python.org/3/library/typing.html) instead of describing the types in the comments.

Reply:
@AlexanderWillner The comments are there for humans. For example, in iPython/Jupyter you can do `things.tasks?` to see the documentation. You'll find this pattern in many common libraries. The particular syntax I followed is the one used by `pandas`, which I find most legible. I believe Wes (who wrote pandas) got it from Travis (numpy), who seems to use a similar syntax.